client/TracyProfiler.cpp

```diff
@@ -913,6 +913,25 @@ struct ProfilerThreadData
 #  endif
 };
 
+#  ifdef TRACY_MANUAL_LIFETIME
+ProfilerData* s_profilerData = nullptr;
+TRACY_API void StartupProfiler()
+{
+    s_profilerData = new ProfilerData;
+    s_profilerData->profiler.SpawnWorkerThreads();
+}
+static ProfilerData& GetProfilerData()
+{
+    while (!s_profilerData);
```

Comment:
Some compilers will warn about this. Add `{}` at the end to silence the warning.

Reply:
Some more thoughts:
- This is essentially a spinlock, there should be some yielding (e.g. `_mm_pause`) between reads, to give memory subsystem some breathing room.
- As this is spinlock, the value read should be atomic. Using relaxed memory ordering will generate identical code to the current loop on x86.
- Is this spin even needed? This is a potential deadlock, which may be hard to diagnose.

Reply:
No, I think it was just left on accident from my earlier experiments, having separate `SpawnWorkerThreads` kind of makes sure that the profiler is ready when other worker threads try to access it.

So at maximum we need an assertion there, if that is a practice you have in your project?

Reply:
Yes, an assert is a clear way of saying that invalid usage is not some random crash due to nullptr.